src/gui/folderitemwidget.cpp

```diff
@@ -253,89 +264,114 @@ void FolderItemWidget::closeFolderView() {
     //    emit displayFolderDetail(_syncDbId, _isExpanded);
 }
 
-void FolderItemWidget::onMenuButtonClicked() {
-    MenuWidget *menu = new MenuWidget(MenuWidget::Menu, this);
 
+SyncInfoClient *FolderItemWidget::getSyncInfoClient() const noexcept {
```

Comment:
I would prefer returning an std::optional<SyncInfoClient &>

Reply:
The C++ compiler won't let me declare any type of the form `std::optional<T&>` for some type `T` (this was my first idea and first attempt). After a second thought, I came to the conclusion that access via pointer is a simple and common interface while it isn't much more risky that access via reference.

Reply:
OK...